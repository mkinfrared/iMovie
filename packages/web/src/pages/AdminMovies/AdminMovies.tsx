import Typography from "@material-ui/core/Typography";
import AddIcon from "@material-ui/icons/Add";
import React, { useCallback } from "react";
import { useHistory } from "react-router-dom";

import AdminMoviesList from "components/AdminMoviesList";
import { AdminRoutes } from "routes/Routes.type";
import FloatButton from "ui/FloatButton/FloatButton";

import css from "./AdminMovies.module.scss";

const AdminMovies = () => {
  const history = useHistory();

  const handleFloatButtonClick = useCallback(() => {
    history.push(AdminRoutes.ADMIN_ADD_MOVIE);
  }, [history]);

  return (
    <>
      <div className={css.AdminMovies}>
        <Typography variant="h3" align="center">
          Admin Movies
        </Typography>
      </div>
      <AdminMoviesList />
      <FloatButton onClick={handleFloatButtonClick}>
        <AddIcon />
      </FloatButton>
    </>
  );
};

export default React.memo(AdminMovies);
