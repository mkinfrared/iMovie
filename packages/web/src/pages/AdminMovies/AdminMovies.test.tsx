import { render } from "@testing-library/react";
import React, { ReactElement } from "react";

import AdminMovies from "./AdminMovies";

jest.mock("react-router-dom", () => ({
  useParams: jest.fn(() => "42"),
  useHistory: jest.fn(() => ({
    goBack: jest.fn(),
    push: jest.fn()
  }))
}));

jest.mock("components/AdminMoviesList", () => (props: any) => (
  <div data-testid="components/AdminMoviesList">
    <code>{JSON.stringify(props)}</code>
  </div>
));

describe("<AdminMovies />", () => {
  let Component: ReactElement;

  beforeEach(() => {
    Component = <AdminMovies />;
  });

  it("should be defined", () => {
    const { container } = render(Component);

    expect(container).toBeDefined();
  });

  it("should match a snapshot", () => {
    const { container } = render(Component);

    expect(container).toMatchSnapshot();
  });
});
