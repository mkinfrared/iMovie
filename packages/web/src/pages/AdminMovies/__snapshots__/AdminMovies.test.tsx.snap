// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<AdminMovies /> should match a snapshot 1`] = `
<div>
  <div>
    AdminMovies
  </div>
  <div
    class="FloatButton"
  >
    <button
      class="MuiButtonBase-root MuiFab-root button MuiFab-primary"
      data-testid="floatButton"
      tabindex="0"
      type="button"
    >
      <span
        class="MuiFab-label"
      >
        <svg
          aria-hidden="true"
          class="MuiSvgIcon-root"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M19 13h-6v6h-2v-6H5v-2h6V5h2v6h6v2z"
          />
        </svg>
      </span>
      <span
        class="MuiTouchRipple-root"
      />
    </button>
  </div>
</div>
`;
